sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -292,7 +293,7 @@ private static Schema fromTableFieldSchema(List<TableFieldSchema> tableFieldSche
       if (!schemaField.getType().getNullable()) {
         field.setMode(Mode.REQUIRED.toString());
       }
-      if (TypeName.ARRAY == type.getTypeName()) {
+      if (type.getTypeName().isCollectionType()) {
```

Comment:
Is there a chance there are other instances of checks like this that need to be replaced?

Reply:
I did another deep search - I found one case in SQL's codegen which I fixed, but can't find anything else.